sdk/trace/provider.go

```diff
@@ -182,6 +182,20 @@ func (p *TracerProvider) ApplyConfig(cfg Config) {
 	p.config.Store(&c)
 }
 
+// Shutdown shuts down the span processors in the order they were registered
+func (p *TracerProvider) Shutdown() {
```

Comment:
Can we change the function signature to require a context be passed and return an error?

i.e.

```suggestion
func (p *TracerProvider) Shutdown(ctx context.Context) error {
```

This would mirror other `Shutdown` methods we have defined (https://github.com/open-telemetry/opentelemetry-go/blob/5660b0b59f4c0fbb9e2879bb645b632b50f5f3bd/sdk/export/trace/trace.go#L47). Additionally, it would allow for the caller to pass a deadline in the context passed, something useful to have so this doesn't block forever and can switch to a more aggressive resource recovery strategy if the deadline is exceeded. Also, the returned error would be useful in communicating failure back up the call chain if one occurs. Otherwise, downstream errors will be dropped or have to use the `ErrorHandler` interface which is less desirable than having the caller make the decision of what to do with an error.

Reply:
Hi, @MrAlias , I updated the signature of Shutdown to include the context in the parameter and return an error. I am not sure if it is necessary to implement checking the deadline is exceed in the method. Could you provide any guidance? Thanks.